src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/UseNameofInPlaceOfString.Fixer.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.CodeQuality.Analyzers.Maintainability
+{
+    /// <summary>
+    /// CA1507 Use nameof to express symbol names
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.VisualBasic, LanguageNames.CSharp), Shared]
+    public class UseNameOfInPlaceOfStringFixer : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(UseNameofInPlaceOfStringAnalyzer.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers'
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(CancellationToken.None).ConfigureAwait(false);
+            var diagnostics = context.Diagnostics;
+            var diagnosticSpan = diagnostics.First().Location.SourceSpan;
+            // getInnerModeNodeForTie = true so we are replacing the string literal node and not the whole argument node
+            var nodeToReplace = root.FindNode(diagnosticSpan, getInnermostNodeForTie: true);
+
+            Debug.Assert(nodeToReplace != null);
+            var stringText = diagnostics.First().Properties[UseNameofInPlaceOfStringAnalyzer.StringText];
```

Comment:
does it has to save that string in diagnostics? rather than get the string from the nodeToReplace? I am asking since holding that in diagnostics is more memory requirement especially if it is only used for fixer and re-acquiring that is not that expensive.

Reply:
ah, sorry, ignore me. getting the stringText is actually expensive since it is actually the symbol the literal is replaced with. okay.

Reply:
actually, looks like this string will be always same as node literal. I think diagnostic should only save the StringText property if argument.ConstatntValue.Value is different than node.ValueText. 

in fixer, if the UserNameofInPlace ... is missing, just use node.ValueText, if it exist, use the one in the property. so that it can save memory in most of common cases.